### PR TITLE
Uniformise the Monoid, Applicative and Monad APIs

### PR DESCRIPTION
**Pull request by jeremiedimino**

### CODE REVIEW

**Review by snowleopard — APPROVED** [review]
Looks nice, thanks!

File: otherlibs/stdune-unstable/top_closure.ml [hunk 1]
```diff
@@ -1,34 +1,34 @@
-module Make (Keys : Top_closure_intf.Keys) (Monad : Monad_intf.S1) = struct
-  open Monad
+module Make (Keys : Top_closure_intf.Keys) (Monad : Monad_intf.S) = struct
```
Comment (snowleopard) [inline otherlibs/stdune-unstable/top_closure.ml:1]: I guess, you could also do this?

```suggestion
module Make (Keys : Top_closure_intf.Keys) (M : Monad) = struct
```

File: otherlibs/stdune-unstable/top_closure.mli [hunk 2]
```diff
@@ -4,6 +4,6 @@ module Int :
 module String :
   Top_closure_intf.S with type key := string and type 'a monad := 'a Monad.Id.t
 
-module Make (Keys : Top_closure_intf.Keys) (Monad : Monad_intf.S1) :
+module Make (Keys : Top_closure_intf.Keys) (Monad : Monad_intf.S) :
```
Comment (snowleopard) [inline otherlibs/stdune-unstable/top_closure.mli:7]: Same here.

File: otherlibs/stdune-unstable/top_closure.ml (hunk 1 above)
Comment (jeremiedimino) [reply otherlibs/stdune-unstable/top_closure.ml:1]: Unfortunately the `Monad` module types doesn't exist inside stdune :(

Comment (snowleopard) [reply otherlibs/stdune-unstable/top_closure.ml:1]: Oh, indeed :(

**Pull request merged**